lib/fizzy/execute.cpp

```diff
@@ -214,8 +226,13 @@ Instance instantiate(const Module& module)
 
 execution_result execute(Instance& instance, FuncIdx function, std::vector<uint64_t> args)
 {
-    // TODO: handle the case when function index points to import
-    const auto& code = instance.module.codesec[function];
+    if (function < instance.imported_functions.size())
+        return instance.imported_functions[function](instance, std::move(args));
```

Comment:
You move args here, but `ImportedFunction` takes args by reference.

Reply:
Yeah, nice catch, I think I'll change `ImportedFunction` to take value

Reply:
I think it still worked correctly due to the rule "reference to temporary extends the lifetime to lifetime of the reference", but it's better not to rely on this obscure rule here, and for symmetry with `execute` pass vector of args by value to `ImportedFunction`, too

Reply:
It was not a bug. The `std::move` is a cast to `&&`, but because of the function argument type is `const&` it becomes `const&&`. The last reference type is not good match for the move constructor, so the copy constructor is called.

Reply:
It copied into a newly constructed temporary indeed